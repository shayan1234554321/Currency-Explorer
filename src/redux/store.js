import { configureStore } from '@reduxjs/toolkit';
import currenciesSlice from './currencies/currenciesSlice';

const store = configureStore({
  reducer: {
    currencies: currenciesSlice,
  },
});

export default store;
