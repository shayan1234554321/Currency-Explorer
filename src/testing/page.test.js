import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import store from '../redux/store';
import Main from '../pages/index';
import SingleCurrency from '../pages/singleCurrency';

describe('Renders without crashing', (() => {
  it('Main renders without crashing', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Main />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('SingleCurrency renders without crashing', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Routes>
            <Route path="/currency" Component={<SingleCurrency />} />
          </Routes>
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
}));
