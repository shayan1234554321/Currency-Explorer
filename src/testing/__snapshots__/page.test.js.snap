// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders without crashing Main renders without crashing 1`] = `
<div
  className="container"
>
  <nav>
    <div
      className="left"
    />
    <h4>
      Base Currency: USD
    </h4>
    <div
      className="iconsContainer"
    >
      <img
        alt="mic"
        className="icon"
        src="mic.svg"
      />
      <img
        alt="setting"
        className="icon"
        src="setting.svg"
      />
    </div>
  </nav>
  <div
    className="topItem"
  >
    <div
      className="flagImageContainer"
    >
      <div
        className="flagImage"
      >
        <img
          alt="flag"
        />
      </div>
    </div>
    <div
      className="details"
    >
      <h2>
         
         
      </h2>
      <h4 />
      <h4 />
    </div>
  </div>
  <div
    className="countryItemsHeader"
  >
    Exchange To Other Countries Currency
  </div>
  <div
    className="countriesContainer"
  />
</div>
`;

exports[`Renders without crashing SingleCurrency renders without crashing 1`] = `null`;
